lib/src/signed_video_h26x_sign.c

```diff
@@ -525,8 +525,7 @@ signed_video_add_nalu_part_for_signing_with_timestamp(signed_video_t *self,
     }
 
     // Finalize GOP hash and generate SEI if the NALU is I frame of a non-empty GOP.
-    if (nalu.is_first_nalu_in_gop && nalu.is_last_nalu_part && gop_info->global_gop_counter > 0) {
-      // Store the timestamp for the first nalu in gop.
+    if (nalu.is_first_nalu_in_gop && nalu.is_last_nalu_part && self->generate_first_sei) {
```

Comment:
I think it would be more readable if you write
```
if (nalu.is_first_nalu_in_gop && nalu.is_last_nalu_part) {
  if (self->generate_first_sei) {
    // The code to generate SEI
  }
  self->generate_first_sei = true;
}
```